### PR TITLE
Issue #3659 jmh tests are creating to many tmp files and never cleaned

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Comment by sbordet** [comment]
@olamy missing signoff?

File: jetty-jmh/src/main/java/org/eclipse/jetty/util/thread/strategy/jmh/EWYKBenchmark.java [hunk 1]
```diff
@@ -89,6 +92,14 @@ public static void setupServer() throws Exception
     @TearDown(Level.Trial)
     public static void stopServer() throws Exception
     {
+        try
+        {
+            IO.delete(directory);
```
Comment (joakime) [inline jetty-jmh/src/main/java/org/eclipse/jetty/util/thread/strategy/jmh/EWYKBenchmark.java]: If we ever want to run these benchmarks on Windows, then this behavior (cleanup / delete at end) will not work (due to file system locks).
Delete on start is usually the better way.

File: jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java [hunk 2]
```diff
@@ -188,23 +188,29 @@ public long testLargeGenerated() throws Exception
                 case "HTTP":
                 {
                     MultiPartFormInputStream parser = new MultiPartFormInputStream(in, _contentType, config, outputDir.toFile());
```
Comment (joakime) [inline jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java]: Is it possible to setup a null/no-op destination for MultiPartFormInputStream?  something that doesn't use a real filesystem?
We want to test the performance of that algorithm, not our filesystem.

File: jetty-jmh/src/main/java/org/eclipse/jetty/util/thread/strategy/jmh/EWYKBenchmark.java (hunk 1 above)
Comment (olamy) [reply jetty-jmh/src/main/java/org/eclipse/jetty/util/thread/strategy/jmh/EWYKBenchmark.java]: sure shouldn't be a problem but this means we will leave a bunch of files.

File: jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java (hunk 2 above)
Comment (olamy) [reply jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java]: will that work on windows if using /dev/null?

File: jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java [hunk 3]
```diff
@@ -177,7 +178,7 @@ public static void setupTrial() throws Exception
     public long testLargeGenerated() throws Exception
     {
         Path multipartRawFile = _file.toPath();
-        Path outputDir = new File("/tmp").toPath();
+        Path outputDir = Paths.get("/dev/null");// Files.createTempDirectory( "jetty_multipart_benchmark");
```
Comment (lachlan-roberts) [inline jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java]: do we need this comment after the line

File: jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java [hunk 4]
```diff
@@ -236,18 +243,16 @@ public long testParser() throws Exception
...
-            //Path multipartRawFile = MavenTestingUtils.getTestResourcePathFile("multipart/" + multiPart + ".raw");
             String expectationPath = "multipart/" + multiPart + ".expected.txt";
-            //Path expectationPath = MavenTestingUtils.getTestResourcePathFile("multipart/" + multiPart + ".expected.txt");
 
             File expectationFile = File.createTempFile( expectationPath, ".tmp" );
 
-            try(InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(expectationPath);
-                OutputStream os = Files.newOutputStream( expectationFile.toPath() )) {
-                IO.copy( inputStream, os );
+            try(InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(expectationPath); //
```
Comment (lachlan-roberts) [inline jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java]: same here

File: jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java [hunk 5]
```diff
@@ -236,18 +243,16 @@ public long testParser() throws Exception
...
+                OutputStream os = Files.newOutputStream( expectationFile.toPath())) {
+                IO.copy(inputStream, os);
             }
 
-            Path outputDir = Files.createTempDirectory( "expected_output_jmh_jetty" );// new File("/tmp").toPath();
+            Path outputDir = Paths.get("/dev/null");//Files.createTempDirectory("expected_output_jmh_jetty");
```
Comment (lachlan-roberts) [inline jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java]: same here

File: jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java (hunk 3 above)
Comment (olamy) [reply jetty-jmh/src/main/java/org/eclipse/jetty/http/jmh/MultiPartBenchmark.java]: just testing the /dev/null with our load server

File: jetty-jmh/src/main/java/org/eclipse/jetty/util/thread/strategy/jmh/EWYKBenchmark.java (hunk 1 above)
Comment (olamy) [reply jetty-jmh/src/main/java/org/eclipse/jetty/util/thread/strategy/jmh/EWYKBenchmark.java]: we cannot really as this directory is created as a tmp file in setup method so we do not know the directory name..

**Pull request merged**